runtime/lua/vim/lsp.lua

```diff
@@ -891,6 +894,55 @@ function lsp.start(config, opts)
   return client_id
 end
 
+--- Consumes the latest progress messages from all clients and formats them as string.
+--- Empty if there are no clients or if no new messages
+---
+---@return string
+function lsp.status()
```

Comment:
We can follow up and add some filters via a `opts` table if it turns out to be required. Given that this function consumes the messages and that common second clients like null-ls probably have no progress messages I'm not sure if this will be required